From 3ede0fcd5b75926beaf7cce163f1743a268e296c Mon Sep 17 00:00:00 2001
From: Hai-Ying Wang <Haiying.Wang@freescale.com>
Date: Tue, 8 May 2012 02:30:48 +0000
Subject: [PATCH 008/987] powerpc/p5040ds: Add USDPAA device tree

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Signed-off-by: Haiying Wang <Haiying.Wang@freescale.com>
(cherry picked from commit ec781fb85d8c1d41c5532d4413c17776cfab59d3)
Change-Id: I42eb939740b1bab3eaa95d8c57078a5bd2d016f9
Reviewed-on: http://git.am.freescale.net:8181/530
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
---
 arch/powerpc/boot/dts/p5040ds-usdpaa.dts | 138 +++++++++++++++++++++++++++++++
 1 file changed, 138 insertions(+)
 create mode 100644 arch/powerpc/boot/dts/p5040ds-usdpaa.dts

diff --git a/arch/powerpc/boot/dts/p5040ds-usdpaa.dts b/arch/powerpc/boot/dts/p5040ds-usdpaa.dts
new file mode 100644
index 0000000..685175c
--- /dev/null
+++ b/arch/powerpc/boot/dts/p5040ds-usdpaa.dts
@@ -0,0 +1,138 @@
+/*
+ * P5040DS Device Tree Source
+ *
+ * Copyright 2012 Freescale Semiconductor Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Freescale Semiconductor nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *
+ * ALTERNATIVELY, this software may be distributed under the terms of the
+ * GNU General Public License ("GPL") as published by the Free Software
+ * Foundation, either version 2 of that License or (at your option) any
+ * later version.
+ *
+ * THIS SOFTWARE IS PROVIDED BY Freescale Semiconductor ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL Freescale Semiconductor BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/include/ "p5040ds.dts"
+
+/ {
+	/* NB: "bpool-ethernet-seeds" is not set to avoid buffer seeding,
+	 * because apps seed these pools with buffers allocated at
+	 * run-time.
+	 * HOWEVER, the kernel driver requires the buffer-size so
+	 * "fsl,bpool-ethernet-cfg" is set. It also mis-interprets
+	 * things if the base-address is zero (hence the 0xdeadbeef
+	 * values).
+	 */
+	bp7: buffer-pool@7 {
+		compatible = "fsl,p5040-bpool", "fsl,bpool";
+		fsl,bpid = <7>;
+		fsl,bpool-ethernet-cfg = <0 0 0 192 0 0xdeadbeef>;
+		fsl,bpool-thresholds = <0x400 0xc00 0x0 0x0>;
+	};
+	bp8: buffer-pool@8 {
+		compatible = "fsl,p5040-bpool", "fsl,bpool";
+		fsl,bpid = <8>;
+		fsl,bpool-ethernet-cfg = <0 0 0 576 0 0xabbaf00d>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	bp9: buffer-pool@9 {
+		compatible = "fsl,p5040-bpool", "fsl,bpool";
+		fsl,bpid = <9>;
+		fsl,bpool-ethernet-cfg = <0 0 0 1600 0 0xfeedabba>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+
+	bp10: buffer-pool@10 {
+		compatible = "fsl,p5040-bpool", "fsl,bpool";
+		fsl,bpid = <10>;
+		fsl,bpool-thresholds = <0x10 0x30 0x0 0x0>;
+	};
+	bp11: buffer-pool@11 {
+		compatible = "fsl,p5040-bpool", "fsl,bpool";
+		fsl,bpid = <11>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+	bp12: buffer-pool@12 {
+		compatible = "fsl,p5040-bpool", "fsl,bpool";
+		fsl,bpid = <12>;
+		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
+	};
+
+	fsl,dpaa {
+		ethernet@2 {
+			compatible = "fsl,p5040-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x54 1 0x55 1>;
+			fsl,qman-frame-queues-tx = <0x74 1 0x75 1>;
+		};
+		ethernet@3 {
+			compatible = "fsl,p5040-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x56 1 0x57 1>;
+			fsl,qman-frame-queues-tx = <0x76 1 0x77 1>;
+		};
+		ethernet@4 {
+			compatible = "fsl,p5040-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x58 1 0x59 1>;
+			fsl,qman-frame-queues-tx = <0x78 1 0x79 1>;
+		};
+		ethernet@5 {
+			compatible = "fsl,p5040-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x5a 1 0x5b 1>;
+			fsl,qman-frame-queues-tx = <0x7a 1 0x7b 1>;
+		};
+		ethernet@8 {
+			compatible = "fsl,p5040-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x60 1 0x61 1>;
+			fsl,qman-frame-queues-tx = <0x80 1 0x81 1>;
+		};
+		ethernet@9 {
+			compatible = "fsl,p5040-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x62 1 0x63 1>;
+			fsl,qman-frame-queues-tx = <0x82 1 0x83 1>;
+		};
+		ethernet@10 {
+			compatible = "fsl,p5040-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x64 1 0x65 1>;
+			fsl,qman-frame-queues-tx = <0x84 1 0x85 1>;
+		};
+		ethernet@11 {
+			compatible = "fsl,p5040-dpa-ethernet-init", "fsl,dpa-ethernet-init";
+			fsl,bman-buffer-pools = <&bp7 &bp8 &bp9>;
+			fsl,qman-frame-queues-rx = <0x66 1 0x67 1>;
+			fsl,qman-frame-queues-tx = <0x86 1 0x87 1>;
+		};
+		dpa-fman0-oh@1 {
+			compatible = "fsl,dpa-oh";
+			/* Define frame queues for the OH port*/
+			/* <OH Rx error, OH Rx default> */
+			fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
+			fsl,fman-oh-port = <&fman0_oh1>;
+		};
+	};
+};
-- 
1.9.1

